lib/ruby2d/window.rb

```diff
@@ -26,14 +27,19 @@ def initialize(args = {})
       @vsync       = args[:vsync]  || true
       @mouse_x, @mouse_y = 0, 0
       @objects     = []
-      @key, @key_down, @key_held, @key_up = [], [], [], []
-      @mouse, @mouse_down, @mouse_up, @mouse_scroll, @mouse_move = [], [], [], [], []
-      @controller, @controller_axis = [], []
-      @controller_button_down, @controller_button_up = [], []
+      @key, @key_down, @key_held, @key_up = {}, {}, {}, {}
+      @mouse, @mouse_down, @mouse_up, @mouse_scroll, @mouse_move = {}, {}, {}, {}, {}
+      @controller, @controller_axis = {}, {}
+      @controller_button_down, @controller_button_up = {}, {}
+      @event_key = "a"
```

Comment:
Looks like overflow doesn't start until after 99,999,999,999,999 in mruby (in mirb, `99999999999999 + 1` returns `1e+14`), which is quite a few events 😉. Probably not a practical concern, but we can keep it as is.